integration-cli/docker_cli_network_unix_test.go

```diff
@@ -221,6 +221,11 @@ func setupRemoteNetworkDrivers(c *check.C, mux *http.ServeMux, url, netDrv, ipam
 	c.Assert(err, checker.IsNil)
 }
 
+func assertNwRemove(c *check.C, nw string) {
```

Comment:
```go
assertNetworkRemoved(c *check.C, network string)
```

Reply:
Hello @dnephin , Thanks for the review.Sincerely!👍  I guess the `change` you mentioned here is not the `assertNetworkRemove()`, because that helper just groups 2 functions together according to @vdemeester 's comment, if I understand correct, the `change` should be this PR itself.  The background of this PR can be traced from the CI result of one of my PRs submitted before: https://jenkins.dockerproject.org/job/Docker-PRs-experimental/39258/consoleFull. If we look at the last 3 failed case, we found that the failure reason is not the test case itself, it's caused by some resources created by previous failed test cases. Another point is, the resource cleanup is not introduced by this PR first in this file, for example, in `TestDockerInspectNetworkWithContainerName()`, we already have this kind of tear-down, but some missed, that's why we have this `enhancement`. If `testEnv.Clean(c)` really can do that, then why we still need this? 
I notice this is a `change requested`, so if I have misunderstand of the `change` context here, then I am happy to make another update only if the `change` requested does make sense.  But if `*_cli_test*` codes will go away at some point, I am also flexible to close this PR and remove it from my `TODO` list.  
haha, at last, forgive my wordy comments🤗

Reply:
> if I understand correct, the change should be this PR itself.

Yes, that's correct

> If `testEnv.Clean(c)` really can do that, then why we still need this?

That is exactly my question. `testEnv.Clean(c)` is supposed to be doing this cleanup. We should figure out why it's not working and fix that, instead of adding more code that duplicates what it should be doing.

Those failures you had are flaky tests, so it's hard for us to say if this change actually fixes them. We would need to be able to reproduce the failures at some rate, and then re-run the suite x times to be sure that they are failing at a lower rate (or not at all).

Where you able to reproduce the failures locally if you ran the tests enough times?